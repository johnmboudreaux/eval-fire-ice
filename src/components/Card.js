import React from 'react';
import PropTypes from 'prop-types';

const Card = ({
  name,
  founded,
  seats,
  titles,
  coatOfArms,
  ancestralWeapons,
  words }) => {

  const checkValue = (houseFounded) => {
    return houseFounded === '' ? 'N/A' : houseFounded;
  };

  return (
    <div className="Card">
      <h2>{name}</h2>
      <p>Founded: {checkValue(founded)}</p>
      <p>Seats:
        {
          seats.map((seat, index) => {
            return (
              <span key={index}>
                {seat}
              </span>
            );
          })
        }
      </p>
      <p>Titles:
        {
          titles.map((title, index) => {
            return (
              <span key={index}>
                {title}
              &nbsp;
              </span>
            );
          })
        }
      </p>
      <p>{coatOfArms}</p>
      <p>Weapons:
        {
          ancestralWeapons.map((weapon, index) => {
            return (
              <span key={index}>
                {weapon}
              </span>
            );
          })
        }
      </p>
      <p>{words}</p>
    </div>
  );
};

Card.propTypes = {
  name: PropTypes.string,
  founded: PropTypes.string,
  seats: PropTypes.array,
  titles: PropTypes.array,
  coatOfArms: PropTypes.string,
  ancestralWeapons: PropTypes.array,
  words: PropTypes.string
};

export default Card;
