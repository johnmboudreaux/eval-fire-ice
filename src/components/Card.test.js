import { shallow } from 'enzyme';
import Card from './Card';
import React from 'react';


describe('Card snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<Card
      name=''
      founded=''
      seats={['']}
      titles={['']}
      coatOfArms=''
      ancestralWeapons={['']}
      words=''
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
