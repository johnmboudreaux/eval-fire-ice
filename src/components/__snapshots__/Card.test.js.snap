// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    ancestralWeapons={
        Array [
            "",
          ]
    }
    coatOfArms=""
    founded=""
    name=""
    seats={
        Array [
            "",
          ]
    }
    titles={
        Array [
            "",
          ]
    }
    words=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <p>
          Founded: 
          N/A
</p>,
        <p>
          Seats:
          <span />
</p>,
        <p>
          Titles:
          <span>
                    
                     
          </span>
</p>,
        <p />,
        <p>
          Weapons:
          <span />
</p>,
        <p />,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "N/A",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "N/A",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats:",
            Array [
              <span />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seats:",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Titles:",
            Array [
              <span>
                
                 
</span>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Titles:",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "",
              " ",
            ],
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Weapons:",
            Array [
              <span />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Weapons:",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <p>
            Founded: 
            N/A
</p>,
          <p>
            Seats:
            <span />
</p>,
          <p>
            Titles:
            <span>
                        
                         
            </span>
</p>,
          <p />,
          <p>
            Weapons:
            <span />
</p>,
          <p />,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "N/A",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "N/A",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats:",
              Array [
                <span />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seats:",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Titles:",
              Array [
                <span>
                  
                   
</span>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Titles:",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "",
                " ",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Weapons:",
              Array [
                <span />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Weapons:",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
